lib/connect/ws.js

```diff
@@ -24,11 +34,15 @@ function buildBuilder (client, opts) {
     wsOpt.protocol = 'mqttv3.1'
   }
 
-  if (opts.wsOptions) {
-    return websocket(url, [wsOpt.protocol], opts.wsOptions)
-  } else {
-    return websocket(url, wsOpt)
+  if (opts.protocol === 'wss') {
```

Comment:
Maybe this doesn't play nice with how `opts.wsOptions` is set in the rest of the code.